### PR TITLE
Added WKT layer to view WKT data on map

### PR DESCRIPTION
**Pull request by sackh**
Signed-off-by: Kharude, Sachin <sachin.kharude@here.com>

@davidbrochart - please review this.

### CODE REVIEW

File: ipyleaflet/leaflet.py [hunk 1]
```diff
@@ -1244,6 +1246,50 @@ def __init__(self, **kwargs):
...
+    ----------
+    path: file path of local WKT file.
+    wkt_string: WKT string.
+    """
+
+    path = Unicode('')
+    wkt_string = Unicode('')
+
+    def __init__(self, **kwargs):
+        super(WKTLayer, self).__init__(**kwargs)
```
Comment (davidbrochart) [inline ipyleaflet/leaflet.py:1264]: Should we start using Python 3's syntax `super().__init__(**kwargs)`?

File: ipyleaflet/leaflet.py [hunk 2]
```diff
@@ -1244,6 +1246,50 @@ def __init__(self, **kwargs):
...
+
+    @observe('path', 'wkt_string', 'style', 'style_callback')
+    def _update_data(self, change):
+        self.data = self._get_data()
+
+    def _get_data(self):
+        if self.path:
+            with open(self.path) as f:
+                wkt_data = f.read()
+                parsed_wkt = wkt.loads(wkt_data)
```
Comment (davidbrochart) [inline ipyleaflet/leaflet.py:1275]: Use `wkt.load(f)` instead?

File: examples/test.wkt [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+-71.1041598612795 42.314808571739,-71.1042515013869 42.3151287620809,
+-71.1041173835118 42.3150739481917,-71.1040809891419 42.3151344119048,
+-71.1040438678912 42.3151191367447,-71.1040194562988 42.3151832057859,
+-71.1038734225584 42.3151140942995,-71.1038446938243 42.3151006300338,
+-71.1038315271889 42.315094347535,-71.1037393329282 42.315054824985,
+-71.1035447555574 42.3152608696313,-71.1033436658644 42.3151648370544,
+-71.1032580383161 42.3152269126061,-71.103223066939 42.3152517403219,
+-71.1031880899493 42.3152774590236)),
+((-71.1043632495873 42.315113108546,-71.1043583974082 42.3151211109857,
+-71.1043443253471 42.3150676015829,-71.1043850704575 42.3150793250568,-71.1043632495873 42.315113108546)))
```
Comment (davidbrochart) [inline examples/test.wkt:23]: Should we start moving data files to a `examples/data` directory?

**Review by davidbrochart — CHANGES_REQUESTED** [review]
Should we add a section in the documentation, as for [GeoJSON](https://ipyleaflet.readthedocs.io/en/latest/api_reference/geo_json.html)?

File: ipyleaflet/leaflet.py [hunk 4]
```diff
@@ -1244,6 +1246,50 @@ def __init__(self, **kwargs):
...
 
+class WKTLayer(GeoJSON):
+    """WKTLayer class.
+
+    Layer created from a local WKT file or WKT string input.
+
+    Attributes
+    ----------
+    path: file path of local WKT file.
+    wkt_string: WKT string.
```
[the remark this answers is not in the document]
Comment (sackh) [reply ipyleaflet/leaflet.py]: sorry, missed this I have updated now.

File: ipyleaflet/leaflet.py (hunk 2 above)
Comment (sackh) [reply ipyleaflet/leaflet.py]: yes, this is a better way, I have updated

File: examples/test.wkt (hunk 3 above)
Comment (sackh) [reply examples/test.wkt]: done

File: ipyleaflet/leaflet.py (hunk 1 above)
Comment (sackh) [reply ipyleaflet/leaflet.py]: done

**Comment by sackh** [comment]
@davidbrochart  - I have updated the docs changes and tested locally. Also did changes as per review comments.

**Comment by davidbrochart** [comment]
Thanks, BTW it looks like `shapely` was already pulled in by a dependency, do you know which one?

**Comment by sackh** [comment]
shapely is a dependency of Geopandas.

**Comment by sackh** [comment]
currently, docs build is failing as input `wkt` file is not committed yet in master once, we merge this it will pass.

File: ipyleaflet/leaflet.py [hunk 5]
```diff
@@ -1244,6 +1246,51 @@ def __init__(self, **kwargs):
...
+            parsed_wkt = wkt.loads(self.wkt_string)
+        else:
+            raise ValueError("Please provide either WKT file path or WKT string")
+
+        geo = geometry.mapping(parsed_wkt)
+        if geo["type"] == "GeometryCollection":
+            feature_collection = {"type": "FeatureCollection", "features": []}
+            for g in geo["geometries"]:
+                feature = {"geometry": g, "properties": {}, "type": "Feature"}
+                feature_collection["features"].append(feature)
```
Comment (davidbrochart) [inline ipyleaflet/leaflet.py:1287]: Maybe:
```python
features = [{"geometry": g, "properties": {}, "type": "Feature"} for g in geo["geometries"]]
feature_collection = {"type": "FeatureCollection", "features": features}
```

Comment (sackh) [reply ipyleaflet/leaflet.py]: done

**Comment by davidbrochart** [comment]
Awesome, thanks!

**Pull request merged**